<!-- -*- mode: XML -*- -->
<launch>
  <arg name="racecar_version" default="$(env CAR_NAME)"  doc="Physical racecar used. Possible values: UNICORNx" />

  <!-- LIDAR -->
  <arg name="lidar_config" default="$(find stack_master)/config/$(arg racecar_version)/devices/lidar.yaml" />
  <node pkg="urg_node" type="urg_node" name="laser_node">
    <rosparam file="$(arg lidar_config)" command="load" />
  </node>

  <!-- IMU -->
  <arg name="imu_config" default="$(find stack_master)/config/$(arg racecar_version)/devices/imu.yaml" />
  <node name="microstrain_inertial_driver" pkg="microstrain_inertial_driver" type="microstrain_inertial_driver_node" output="screen">
    <rosparam file="$(arg imu_config)" command="load" />
  </node>

  <!-- VESC -->
  <group ns="vesc">
    <include file="$(find stack_master)/launch/vesc.launch">
      <arg name="racecar_version" value="$(arg racecar_version)" />
    </include>

    <node pkg="stack_master" type="simple_mux.py" name="simple_mux" output="screen"/>
  </group>

  <include file="$(find stack_master)/config/$(arg racecar_version)/devices/static_transforms.launch.xml"/>

  <node pkg="stack_master" type="relay_node.py" name="relay_node" output="screen"/>

  <node pkg="joy" type="joy_node" name="joy_node" output="screen"/>
  <!-- <node pkg="joy" type="joy_node" name="joy_node"/> -->

</launch>
